content_scripts/mode_find.js

```diff
@@ -266,7 +266,12 @@ class FindMode extends Mode {
   }
 
   // Returns null if no search has been performed yet.
-  static getQuery(backwards) {
+  static getQuery(backwards, query) {
+    if (query) {
+      this.query = {};
+      this.updateQuery(query);
+      this.saveQuery();
+    }
     if (!this.query) return;
```

Comment:
**If** we don't want to replace `this.query`, I think these changes **might** warrant revising this function and all of its uses to instead use the passed in `query` parameter rather than `this.query`. Otherwise, we can avoid passing a `query` parameter by updating `this.query` elsewhere (as explained at the end).
If we always pass query in, when we are calling it with a `n` or `N` command, we can pass in the `this.query` query text, but when we call it with `*` or `#` we can pass in our selection.
This would also solve a (**potential**) bug in the current implementation where if we start a search query, (example: `/foo`), then select a word and search it (example: select `bar` and go next `*`), it replaces our search query so that pressing `n` will go to the next instance of "bar", **not** the next instance of "foo". This ***is** the default behavior of Vim*, so it is probably what we want in this case. However, if we want to implement a command in the future that searches but *shouldn't replace the query string*, this will cause a bug when we try to use this function. It seems like it might be a good idea to not bake the query replacing logic into `getQuery` for this reason.
Even given we want to keep this behavior (I think we probably should), we can still refactor this. In that case, I would suggest removing the query parameter here and just setting `FindMode.query` when we call it in `normal_mode.js`.